ml-proto/spec/int.ml

```diff
@@ -208,7 +210,49 @@ struct
   let ge_s x y = x >= y
   let ge_u x y = cmp_u x (>=) y
 
-  let of_string = Rep.of_string
+  let parse_hexdigit c =
+    int_of_char c -
+      if '0' <= c && '9' >= c then 0x30
+      else if 'a' <= c && 'f' >= c then 0x61 - 10
+      else if 'A' <= c && 'F' >= c then 0x41 - 10
+      else failwith "unexpected digit"
+
+  let neg_safe x =
+    if lt_s (sub x Rep.one) Rep.minus_one then raise Numerics.IntegerOverflow;
+    Rep.neg x
+
+  (* This implementation allows leading signs and unsigned values *)
+  let of_string x =
+    let open Rep in
+    let len = String.length x in
+    let require b = if not b then failwith "of_int" in
```

Comment:
Everytime I call "require" I want it to be the same error message "of_string", but if I move it out, I can't hardcode this message, since this particular error message does only make sense in the scope of my "of_string"-function. I will have to supply the same error string everytime I call the hoisted "require"-function. This seemed a bit redundant to me. This is no big deal, I guess, I simply thought, that if I only have one error message, my solution would be a bit compact.